pkg/controller/jobframework/setup.go

```diff
@@ -99,6 +105,40 @@ func (m *integrationManager) setupControllers(mgr ctrl.Manager, log logr.Logger,
 	})
 }
 
+func (m *integrationManager) setupControllerAndWebhook(mgr ctrl.Manager, name string, fwkNamePrefix string, cb IntegrationCallbacks, options Options, opts ...Option) error {
+	if err := cb.NewReconciler(
+		mgr.GetClient(),
+		mgr.GetEventRecorderFor(fmt.Sprintf("%s-%s-controller", name, options.ManagerName)),
+		opts...,
+	).SetupWithManager(mgr); err != nil {
+		return fmt.Errorf("%s: %w", fwkNamePrefix, err)
+	}
+	if err := cb.SetupWebhook(mgr, opts...); err != nil {
+		return fmt.Errorf("%s: unable to create webhook: %w", fwkNamePrefix, err)
+	}
+	m.enableIntegration(name)
+	return nil
+}
+
+func (m *integrationManager) waitForAPI(ctx context.Context, mgr ctrl.Manager, log logr.Logger, gvk schema.GroupVersionKind, action func()) {
+	for {
+		m.mu.Lock()
+		_, err := mgr.GetRESTMapper().RESTMapping(gvk.GroupKind(), gvk.Version)
+		m.mu.Unlock()
+		if err != nil {
```

Comment:
Make sure the error is logged. Can you paste here an example log line?

Reply:
I hesitated on adding a log here as it would log for each API that is not yet available, and will retry after the specified interval. - This could cause log flooding. The error serves here as an indication to retry.

I've refactored the `waitForAPI()` to reflect explicitly the expected behaviour. - Let me know what you think, thank you.

Reply:
Can we log only unexpected errors?

Reply:
Absolutely, I've added an error log for unexpected errors, such as any error that is not a `IsNoMatchError`